snap/info.go

```diff
@@ -1453,9 +1453,19 @@ func (app *AppInfo) SecurityTag() string {
 func (app *AppInfo) DesktopFile() string {
 	desktopFileIDs, err := app.Snap.DesktopPlugFileIDs()
 	if err != nil || len(desktopFileIDs) == 0 {
-		// fallback to a simple heuristic "$PREFIX_$APP.desktop"
-		return filepath.Join(dirs.SnapDesktopFilesDir, fmt.Sprintf("%s_%s.desktop", app.Snap.DesktopPrefix(), app.Name))
+		return app.fallbackDesktopFile()
 	}
+
+	if app.CommonID != "" {
+		desktopID := strings.TrimSuffix(app.CommonID, ".desktop")
+
+		if !strutil.ListContains(desktopFileIDs, desktopID) {
+			return app.fallbackDesktopFile()
+		}
+
+		desktopFileIDs = []string{desktopID}
```

Comment:
the `.destkop` suffix has been dropped, this this will need be updated if/when #15811 lands which assumes the suffix is still there